src/main/groovy/wooga/gradle/snyk/tasks/SnykInstall.groovy

```diff
@@ -14,21 +16,45 @@ import wooga.gradle.snyk.tasks.internal.SnykDownloadAsset
 
 class SnykInstall extends DefaultTask implements SnykInstallSpec {
 
-    private static final String SNYK_RELEASE_JSON_URL = "https://static.snyk.io/cli/%s/release.json"
+    static String getReleaseJsonUrl(String version) {
+        def scheme = System.getProperty("static.snyk.io.scheme", "https")
+        def port = System.getProperty("static.snyk.io.port", "443")
+        def host = System.getProperty("static.snyk.io.host", "static.snyk.io")
+        "${scheme}://${host}:${port}/cli/${version}/release.json"
+    }
 
     @OutputFile
     final Provider<RegularFile> snykExecutable
 
     private final Provider<Map<String, SnykDownloadAsset>> snykReleasesProvider
 
-    SnykInstall() {
-        this.snykReleasesProvider =  version.map ({
-            String version -> fetchSnykReleases(version)
-        }.memoize())
+    private final Provider<String> versionToInstall
+
+    @Input
+    Provider<String> getVersionToInstall() {
+        versionToInstall
+    }
 
+    SnykInstall() {
         this.snykExecutable = installationDir.zip(executableName) {
             dir, fileName -> dir.file(normalizeExecutableByOS(fileName))
         }
+
+        this.versionToInstall = version.map({
+            def version = it
+            if (version == "latest") {
+                def releasesURL = getReleaseJsonUrl(version)
```

Comment:
I guess we can put these three lines in a method, something like `fetchLatestVersionNumber`, I guess?

Reply:
Yes. But I like to do this only if I see a reason to create multiple methods.